ptsemseg/loss.py

```diff
@@ -1,11 +1,25 @@
 import torch
 import numpy as np
+import scipy.misc as m
 import torch.nn as nn
 import torch.nn.functional as F
 
+from torch.autograd import Variable
+
 
 def cross_entropy2d(input, target, weight=None, size_average=True):
     n, c, h, w = input.size()
+    nt, ht, wt = target.size()
+
+    if h != ht or w != wt: # inconsistent size between input and target
+        lbl = target.data.cpu().numpy()
+        lbl = lbl.astype(float)
+        lbl_resized = np.zeros((n, h, w))
+        for i in range(nt):
+            lbl_resized[i,:,:] = m.imresize(lbl[i,:,:], (h, w), 'nearest', mode='F')
+        lbl_resized = lbl_resized.astype(int)
+        target = Variable(torch.from_numpy(lbl_resized).long().cuda())
```

Comment:
Perhaps it would be faster if we used `torch.nn.functional.upsample_nearest` instead of scipy `imresize`.